native/membrane_videocompositor_opengl_rust/src/lib.rs

```diff
@@ -0,0 +1,246 @@
+#![deny(unsafe_op_in_unsafe_fn)]
+//! This crate is a video compositor implementation using OpenGL, intended for use with an Elixir package. 
+
+extern crate khronos_egl as egl;
+
+use glad_gles2::gl;
+use rustler::ResourceArc;
+
+pub mod framebuffers;
+pub mod scene;
+pub mod shaders;
+pub mod textures;
+
+#[allow(non_snake_case)]
+mod atoms {
+  rustler::atoms! {
+    ok,
+    error,
+    test_atom,
+    unsupported_pixel_format,
+    I420,
+
+    egl_not_initialized,
+    egl_bad_access,
+    egl_bad_alloc,
+    egl_bad_attribute,
+    egl_bad_context,
+    egl_bad_config,
+    egl_bad_current_surface,
+    egl_bad_display,
+    egl_bad_surface,
+    egl_bad_match,
+    egl_bad_parameter,
+    egl_bad_native_pixmap,
+    egl_bad_native_window,
+    egl_context_lost,
+  }
+}
+
+#[derive(Debug, rustler::NifStruct)]
```

Comment:
FYI that works

```suggestion
use rustler::NifStruct;

#[derive(Debug, NifStruct)]
```

Reply:
I know, I just like leaving a 'namespace' in places like this, rather than having a use for the entire module.

Reply:
do as you wish :) this was just a hint